lib/std/mem.zig

```diff
@@ -959,15 +958,14 @@ test "len" {
     try testing.expect(len(c_ptr) == 2);
 }
 
+/// Returns the index of the sentinel from the sentinel-terminated pointer.
 pub fn indexOfSentinel(comptime Elem: type, comptime sentinel: Elem, ptr: [*:sentinel]const Elem) usize {
     var i: usize = 0;
-    while (ptr[i] != sentinel) {
-        i += 1;
-    }
+    while (ptr[i] != sentinel) : (i += 1) {}
```

Comment:
What's the point of this change? There's nothing hugely *wrong* with writing it like this, but again, it just feels like using language features for the sake of using them. Every while loop has a body, but not every while loop has a post-expression, so the null hypothesis is to not use a post-expression unless it improves the code.

Reply:
Thanks for the review.
For example,
```zig
pub fn trimLeft(comptime T: type, slice: []const T, values_to_strip: []const T) []const T {
    var begin: usize = 0;
    while (begin < slice.len and indexOfScalar(T, values_to_strip, slice[begin]) != null) : (begin += 1) {}
    return slice[begin..];
}
```
This function has already been there with empty while loop body, so adhering to that I changed that. So should i revert it?

Reply:
Hm, interesting. Personally, yes, I'd change the other use site to use the loop body rather than a post-expression. Post-expressions are effectively syntax sugar for common looping patterns, and when you don't need to use that sugar it just seems like an unnecessary complication. Also, note that nothing stops you from putting the loop on one line, as in `while (cond) i += 1;`.

Reply:
Thats cool, I sticked with that and converted all the empty while loop bodies with their continue exp as their bodies.